caffe2/operators/relu_op.cc

```diff
@@ -29,8 +29,9 @@ bool ReluOp<float, CPUContext>::RunOnDevice() {
 
 template <>
 bool ReluGradientOp<float, CPUContext>::RunOnDevice() {
-  auto& Y = Input(0);
-  auto& dY = Input(1);
+  auto& X = Input(0);
```

Comment:
Ok, I'll be very explicit here. Relu gradient *should* be using X, the definition of it requires X. However, your CPU implementation doesn't, so I'll comment out this line, although it'll look weird starting with Input(1).